monitoring/db_update_sqlite.py

```diff
@@ -318,13 +378,116 @@ def refresh_BenchmarksBySubmitHost_from_view(view_name):
+            FROM VSyncRecords
+            WHERE
+                Year >= YEAR(NOW()) - 2
+            GROUP BY Site, Year, Month, SubmitHost;
+        """
+        fetchset_syncrecords = VSyncRecords.objects.using('grid').raw(sql_query_syncrec)
+
+        # Build dicts using helper methods
+        summaries_dict = summaries_dict_standard.copy()
+        syncrecords_dict = syncrecords_dict_standard.copy()
+
+        for row in fetchset_summaries:
+            summaries_dict = fill_summaries_dict(summaries_dict, row)
+
+        for row in fetchset_syncrecords:
+            syncrecords_dict = fill_syncrecords_dict(syncrecords_dict, row)
+
+        summaries_dict = correct_dict(summaries_dict)
+        syncrecords_dict = correct_dict(syncrecords_dict)
+
+        # Convert to DataFrames
+        df_summaries = pd.DataFrame.from_dict(summaries_dict)
+        df_syncrecords = pd.DataFrame.from_dict(syncrecords_dict)
+
+        # Merge on all keys
+        df_all = df_summaries.merge(
+            df_syncrecords,
+            left_on=['Site', 'Month', 'Year', 'SubmitHostSumm'],
+            right_on=['Site', 'Month', 'Year', 'SubmitHostSync'],
+            how='outer'
+        )
+
+        # Store in the local DB
+        for record in df_all.to_dict('records'):
+            site = record.get("Site")
+            month = record.get("Month")
+            year = record.get("Year")
+            submit_host = record.get("SubmitHostSumm") or record.get("SubmitHostSync")
+            record_start = record.get("RecordStart")
+
+            # Skip rows where RecordStart is missing or NaN
+            if pd.isna(record_start):
+                continue
+
+            # Sanitize numeric fields
+            record_count_published = record.get("RecordCountPublished")
+            if pd.isna(record_count_published):
+                record_count_published = 0
+
+            record_count_in_db = record.get("RecordCountInDb")
+            if pd.isna(record_count_in_db):
+                record_count_in_db = 0
+
+            GridSiteSyncSubmitH.objects.update_or_create(
+                SiteName=site,
+                YearMonth=get_year_month_str(year, month),
+                Month=month,
+                Year=year,
+                SubmitHost=submit_host,
+                defaults={
+                    'RecordStart': record_start,
+                    'RecordEnd': record.get("RecordEnd"),
+                    'RecordCountPublished': record_count_published,
+                    'RecordCountInDb': record_count_in_db,
+                }
+            )
+
+        log.info("Refreshed GridSiteSyncSubmitH")
+
+    except DatabaseError:
+        log.exception("Error while trying to refresh GridSiteSyncSubmitH")
+
+
 if __name__ == "__main__":
     log.info('=====================')
 
     refresh_gridsite()
     refresh_cloudsite()
     refresh_gridsitesync()
     refresh_BenchmarksBySubmitHost()
+    refresh_gridsitesync_submithost()
```

Comment:
I'd like these ordered from fastest to slowest so that fresh data appears sooner.

Do you know how long the new function takes?